fastpool/src/unbounded.rs

```diff
@@ -231,11 +231,53 @@ where
     }
 }
 
+// Methods for `Pool` with `NeverManageObject`.
 impl<T: Send + Sync> Pool<T> {
     /// Creates a new [`Pool`] from config and the [`NeverManageObject`].
     pub fn from_config(config: PoolConfig) -> Arc<Self> {
         Self::new(config, NeverManageObject::<T>::default())
     }
+
+    /// Retrieves an [`Object`] from this [`Pool`], or creates a new one with the passed-in async
+    /// closure, if the pool is empty.
+    ///
+    /// This method should be called with a pool wrapped in an [`Arc`].
+    ///
+    /// This method only exists for [`NeverManageObject`] pools. If you provide a custom
+    /// [`ManageObject`] implementation, you should use [`Pool::get`] instead, and it will call
+    /// [`ManageObject::create`] to create a new object if the pool is empty.
+    pub async fn get_or_create<E, F>(self: &Arc<Self>, f: F) -> Result<Object<T>, E>
```

Comment:
Could be implemented like:

```rust
let o = match pool.get().await {
  Ok(o) => o,
  Err(err) => { pool.put(f().await?); pool.get().await.unwrap() }
}
```

But there can be others get the just put object before re-get.